src/libcollections/vec_deque.rs

```diff
@@ -2121,6 +2122,42 @@ impl<T: fmt::Debug> fmt::Debug for VecDeque<T> {
     }
 }
 
+#[stable(feature = "rust1", since = "1.9.0")]
+impl<T> From<Vec<T>> for VecDeque<T> {
+    fn from(other: Vec<T>) -> Self {
+        unsafe {
+            let other_buf: *mut T = mem::transmute(other.as_ptr());
```

Comment:
If you declare `other` to be `mut`, this can just be:

```rust
let other_buf = other.as_mut_ptr();
```

(The transmute is unnecessary.)